tinkerbell/stack/templates/_hook.tpl

```diff
@@ -0,0 +1,14 @@
+{{- define "stack.hookDownloadJobEnabled" -}}
```

Comment:
can we get a code comment here about what this is and why? Reading the code is a little hard to follow and will require that I re-understand it every time I come back to it. Also, i don't know if I understand it in the first place.

Is this logic to determine if downloading HookOS should occur?
Does this logic say that if there is a pvc with accessMode of ReadWriteOnce then we won't download HookOS? if so, why is that?

Reply:
> Is this logic to determine if downloading HookOS should occur?

Yes

> Does this logic say that if there is a pvc with accessMode of ReadWriteOnce then we won't download HookOS? if so, why is that?

Because you can't determine on which node will be scheduled the Kubernetes Job, so stack Deployment resource already mounting the PVC, the download job will fail to mount the PVC